src/main/java/com/microsoft/aad/msal4j/TokenRequestExecutor.java

```diff
@@ -80,6 +76,10 @@ OAuthHttpRequest createOauthHttpRequest() throws SerializeException, MalformedUR
                         .createClientAuthFromClientAssertion((ClientAssertion) ((ClientCredentialRequest) msalRequest).parameters.clientCredential())
                         .applyTo(oauthHttpRequest);
             } else {
+                Map<String, List<String>> queryParameters = oauthHttpRequest.getQueryParameters();
```

Comment:
It looks like client ID would only be added to the request if the request has a client assertion (the main 'if' block, if (msalRequest.application().clientAuthentication() != null)), but not if a client assertion was set as a parameter for the request (the nested if/else, if (msalRequest instanceof ClientCredentialRequest && ... ).

Is some behavior different when a developer wants a to override their client app's assertion for a certain request? If not, then this new code should probably be in the main 'if' block instead of the nested 'if/else' block.

Reply:
makes sense. I have updated the PR.